src/medCore/gui/toolboxes/medAbstractSelectableToolBox.cpp

```diff
@@ -0,0 +1,67 @@
+/*=========================================================================
+
+ medInria
+
+ Copyright (c) INRIA 2013 - 2014. All rights reserved.
+ See LICENSE.txt for details.
+ 
+  This software is distributed WITHOUT ANY WARRANTY; without even
+  the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+  PURPOSE.
+
+=========================================================================*/
+
+#include <medAbstractSelectableToolBox.h>
+#include <medSelectorToolBox.h>
+#include <medTabbedViewContainers.h>
+
+class medAbstractSelectableToolBoxPrivate
+{
+public:
+    medSelectorToolBox * selectorToolBox;
+};
+
+//! Parent should be a medSelectorToolBox
+medAbstractSelectableToolBox::medAbstractSelectableToolBox(QWidget *parent) : medToolBox(parent), d(new medAbstractSelectableToolBoxPrivate)
+{
+    d->selectorToolBox = qobject_cast<medSelectorToolBox*>(parent);
+}
+
+medAbstractSelectableToolBox::~medAbstractSelectableToolBox(void)
+{
+    delete d;
+    d = NULL;
+}
+
+void medAbstractSelectableToolBox::setSelectorToolBox(medSelectorToolBox *toolbox)
+{
+    d->selectorToolBox = toolbox;
+}
+
+medSelectorToolBox *medAbstractSelectableToolBox::selectorToolBox(void)
+{
+    return d->selectorToolBox;
+}
+
+void medAbstractSelectableToolBox::showEvent(QShowEvent *event)
```

Comment:
We definitely have to rm `behaveWith...` since with this, it seems that when you show a tbx, it runs this `showEvent()` and `behaveWith...`